test/functional/autocmd/textyankpost_spec.lua

```diff
@@ -0,0 +1,166 @@
+local helpers = require('test.functional.helpers')
+local clear, eval, eq, insert = helpers.clear, helpers.eval, helpers.eq, helpers.insert
+local feed, execute, expect = helpers.feed, helpers.execute, helpers.expect
+
+
+describe('TextYankPost', function()
+  before_each(function()
+    clear()
+
+    -- emulate the clipboard so system clipboard isn't affected
+    execute('let &rtp = "test/functional/fixtures,".&rtp')
+
+    execute('let g:count = 0')
+    execute('autocmd TextYankPost * let g:event = copy(v:event)')
+    execute('autocmd TextYankPost * let g:count += 1')
+  end)
+
+  it('is executed after yank and handles register types', function()
+    insert([[
+      foo bar
+      baz text]])
```

Comment:
Test text should contain `NUL` (to check that result is really readfile()-style list of lines because here `NUL` is transformed into `NL`). I would suggest writing this as

```Lua
curbufmeths.set_line_slice(0, 0, true, true, {
  'foo\0bar',
  'baz text',
})
```

and put this into `before_each` function, you in any case set buffer text to two lines *always*, what’s the point of having these lines be different?

(BTW, @tarruda set_line_slice is not tested for it throwing any errors at all and not tested for NUL handling. Also why “string cannot contain newlines” is Exception and not Validation?)